tests/unit/test_kernel.py

```diff
@@ -202,22 +222,371 @@ def expected_divergence_x_grad_y(
         """Compute expected divergence of the kernel w.r.t ``x`` gradient ``y``."""
 
 
+class KernelCompositionTest(ABC, Generic[_Kernel]):
```

Comment:
Wasn't sure how to make this test class Generic on _AdditiveKernel and _ProductKernel

Reply:
Or if we want that